### PR TITLE
Add healthchecks to container-compose.yml

### PR DESCRIPTION
**Pull request by RaphC**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This PR replaces dockerize entrypoint used to check dependency availibility (redis, db, rails) by the healthcheck feature added in compose. No need to copy db, redis port information and so one, no need to use an extra entrypoint, conditions are more precise.

## Added/updated tests?

- [ ] Yes
- [x]  No, and this is why: No code changes, only the compose file
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Comment by rt4914** [comment]
Hi @RaphC 
We really appreciate your efforts on this PR. Is this PR ready for review? If yes, can you mark it as `Ready for reivew` to take the PR ahead. 
Screenshot 2023-04-05 at 11 34 22 PM

**Review by maestromac — APPROVED** [review]
Thank you for this neat change. LGTM!

You are welcome to apply the same changes to our `docker-compose.yml` if you would like. Can be on the same or another PR.

File: container-compose.yml [hunk 1]
```diff
@@ -83,16 +92,18 @@ services:
...
     depends_on:
-      - rails
-      - redis
-      - db
+      rails:
+        condition: service_healthy
+      redis:
+        condition: service_healthy
+      db:
+        condition: service_healthy
```
Comment (klardotsh) [inline container-compose.yml:106]: Due to the use of this `depends_on.[*].condition`, which exists only in v2.4+ and in Compose Spec (but *not* any of the 3.x formats), and due to an above use of `start_period` (which exists in 2.1+, 3.4+, and Compose Spec), let's go ahead and remove the `version: "3"` from the top of this file: there is no v3 format that can make this file valid, and [`version` is deprecated anyway](https://docs.docker.com/compose/compose-file/04-version-and-name/) so there's no point in fixing this anomaly.

File: container-compose.yml [hunk 2]
```diff
@@ -101,12 +112,16 @@ services:
...
     volumes:
       - .:/opt/apps/forem:z
-    entrypoint: ["dockerize", "-wait", "tcp://db:5432", "-wait", "tcp://redis:6379", "-wait", "http://rails:3000", "-timeout", "300s"]
     command: ["bundle", "exec", "sidekiq","-c","2"]
 
   db:
     image: postgres:11-alpine
     container_name: forem_postgresql
+    healthcheck:
+      test: [ "CMD-SHELL", "pg_isready -d forem -U forem" ]
```
Comment (klardotsh) [inline container-compose.yml:121]: Does this need `CMD-SHELL`? Can it use `CMD` like `redis-cli` does below?

File: container-compose.yml [hunk 3]
```diff
@@ -10,6 +10,12 @@ services:
       - db
       - redis
       - yarn
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f http://localhost:3000/ || exit 1"]
```
Comment (klardotsh) [inline container-compose.yml:14]: - Does this need `CMD-SHELL`? Can it use `CMD` like `redis-cli` does below?
- Can we get away without changing the return code here? `curl(1)` is already pretty good about returning a myriad of non-zero error codes in the event of failure, and I'm not aware of a reason to coalesce them all to `1`. See for example:

```sh
(woods) ~  » curl foo.bar.example.com; echo $?
curl: (6) Could not resolve host: foo.bar.example.com
6
```

**Review by klardotsh — CHANGES_REQUESTED** [review]

File: container-compose.yml (hunk 1 above)
Comment (RaphC) [reply container-compose.yml:106]: Hi @klardotsh you absolutely right, I'll remove the version attribute

File: container-compose.yml (hunk 2 above)
Comment (RaphC) [reply container-compose.yml:121]: Sure we can. It should be better even as a container not always has a SHELL (/bin/shell)  prompt. I'll fix it.

File: container-compose.yml (hunk 3 above)
Comment (RaphC) [reply container-compose.yml:14]: The same :). I will switch to a CMD.

**Pull request merged**